### PR TITLE
Fix publishing from Travis CI

### PR DESCRIPTION
**Pull request by marcphilipp**
When deploying a release to Nexus, it implicitly creates a staging
repository based on the uploader's IP address. However, when uploading
from a Travis CI build, the external IP address is not stable and thus
Nexus creates multiple staging repositories that can neither be merged
nor released separately in the end.

This commit introduces the `de.marcphilipp.nexus-publish` Gradle plugin
that explicitly creates a staging repository prior to publishing and
uses a URL specific to this repository for uploading. Thus, the auto-
staging feature of Nexus is not used and we're no longer subject to
Travis' issue with external IP addresses.

---
This change is [Reviewable](https://reviewable.io/reviews/spockframework/spock/952)

### CODE REVIEW

**Comment by szpak** [comment]
@leonard84 When do you plan to remove support for Java 7 in Spock?

**Comment by leonard84** [comment]
Hm, I wanted to drop 7 with Spock 2.x, so after 1.3 ;)

**Comment by leonard84** [comment]
IIRC setting `targetCompatibility` does not ensure binary/library compatibility, you'd need to set another parameter for the correct classpath. This was only fixed with the `--release` option in JDK 9+. Although it is currently a PITA I think it would be best to wait with the merge until after 1.3, each other option does add work for a temporary workaround.

**Comment by marcphilipp** [comment]
@leonard84 Shall I change this PR to target the `spock-2.0` branch?

**Comment by leonard84** [comment]
@marcphilipp yeah

**Comment by marcphilipp** [comment]
Done.

@leonard84 PTAL!

File: build.gradle [hunk 1]
```diff
@@ -6,6 +6,7 @@ plugins {
   id "org.asciidoctor.convert" version "1.5.9.2"
   id "jacoco"
   id "net.nemerosa.versioning" version "2.8.2"
+  id "de.marcphilipp.nexus-publish" version "0.1.0" apply false
```
Comment (szpak) [inline build.gradle]: Why not 0.1.1?

Comment (marcphilipp) [reply build.gradle]: Because this PR is very old... 😉 

I'll change it.

Comment (marcphilipp) [inline build.gradle]:
```suggestion
  id "de.marcphilipp.nexus-publish" version "0.1.1" apply false
```

**Pull request merged**